src/cli/add.rs

```diff
@@ -73,6 +83,29 @@ pub struct Args {
     /// Whether the pypi requirement should be editable
     #[arg(long, requires = "pypi")]
     pub editable: bool,
+
+    /// Where to add the pypi requirement to the manifest
+    #[arg(long, requires = "pypi")]
```

Comment:
This location thing works but confused me a little as it is also required in combination with `--feature` 
What if we instead extend the CLI with `--optional-dependencies [OPTIONAL_GROUP]` `--dependency-group [DEPENDENCY_GROUP]` and only force it into `tool.pixi` if it has a `--editable`  or `--platform`

Then it would result in this:
```
> pixi add --pypi numpy
> pixi add --pypi --optional-dependencies cuda torch
> pixi add --pypi --dependency-group test pytest
> pixi add --pypi "package @ PATH" --editable
> pixi add --pypi --platform linux-64 cmake
```
```toml
[project]
dependencies = ["numpy>=2.1.3,<3"]

[project.optional-dependencies]
cuda = ["torch"]

[dependency-groups]
test = ["pytest>=8.3.3,<9"]

[tool.pixi.pypi-dependencies]
package = { path = "PATH", editable = true }

[tool.pixi.target.linux-64.pypi-dependencies]
cmake = "*"
```

Reply:
Yes, that may be a better option. Two questions @ruben-arts 

- This would mean one could not force the dependency to be added to the `tool.pixi.pypi-dependencies` table; Is that OK? Note that with this PR, it is possible, for both default or non-default features
- What would happen if one specifies a `—feature` ?

Reply:
Features are pixi only so lets make that the `tool.pixi.feature.name.pypi-dependencies`.
I don't have an opinionated answer for the first question but we could just allow `--feature default` to force it into `tool.pixi.pypi-dependencies`.
I added it to this initial post, let me know what you think.

Reply:
Hum I see. I am a bit hesitant on this. I was thinking it would be better to have a single CLI API (i.e. via the --feature), and write the dependency in the most logical place, depending on the manifest type (and the editable / platform args).

A short term option may be to not modify the CLI API at all, and see what happens / what users require / think / need once it is released and pypi dependencies flow to dependency-groups ?

Reply:
That would make sense for now.

Reply:
See https://github.com/prefix-dev/pixi/issues/2493 @ruben-arts